lms/djangoapps/course_api/api.py

```diff
@@ -268,3 +269,65 @@ def get_course_run_url(request, course_id):
     """
     course_run_url = reverse('openedx.course_experience.course_home', args=[course_id])
     return request.build_absolute_uri(course_run_url)
+
+
+def get_course_members(course_key, include_students=True, access_roles=None, prefetch_accessroles=False, prefetch_enrollments=False):
+    """
+    Returns a User queryset that filters all users related to a course.
+    For example - Students, Teachers, Staffs etc.
+
+    Arguments:
+        course_key (CourseKey): the CourseKey for the course
+        include_students: Wether or not to include students,
+        access_roles:
+            accepts an array of string course access roles.
+            If None provided, it includes all roles.
+        prefetch_accessroles:
+            prefetches CourseAccessRole instances attached with user.
+            This only includes CourseAccessRole instances related to provided CourseKey,
+        prefetch_enrollments:
+            prefetches CourseEnrollment instances attached with user.
+            This only includes CourseEnrollment instances related to provided CourseKey,
+    """
+    queryset = User.objects.filter()
+
+    # if access_roles not given, assign all registered access roles
+    if access_roles is None:
+        access_roles = REGISTERED_ACCESS_ROLES.keys()
+
+    # conditions for filtering based on CourseAccessRole
+    access_role_qs = Q(
+        courseaccessrole__course_id=course_key,
+        courseaccessrole__role__in=access_roles
+    )
+
+    # conditions for filtering based on CourseEnrollment
+    students_qs = Q(
+        courseenrollment__course_id=course_key,
+        courseenrollment__is_active=True
+    )
+
+    if include_students:
+        queryset = queryset.filter(access_role_qs | students_qs)
+    else:
+        queryset = queryset.filter(access_role_qs)
+
+    # prefetch CourseAccessRole items related to the course and given roles
+    if prefetch_accessroles:
+        queryset = queryset.prefetch_related(Prefetch(
+            'courseaccessrole_set',
+            CourseAccessRole.objects.filter(
+                course_id=course_key, role__in=access_roles)
+        ))
+
+    # prefetch CourseEnrollment items related to the course
+    if prefetch_enrollments:
+        queryset = queryset.prefetch_related(Prefetch(
+            'courseenrollment_set',
+            CourseEnrollment.objects.filter(course_id=course_key)
+        ))
```

Comment:
I think we can remove the prefetching here and do it in the viewset, inside the `get_queryset method`.
This needs to be a simple API to retrieve users enrolled into a course depending on their role.

Reply:
Yeah, we can. But in that case, we would get all the ``CourseEnrollment`` of the user, not only those related to the Course. In order to do that, we would need to import ``CourseEnrollment`` in ``lti_consumer.compat`` module. Should we do that and remove prefetching here?